File: wsm-web/src/api/platform.ts
import request from './request'
import type { ApiResponse, PageParams, PageResult } from './request'

export interface Platform {
  id: number
  name: string
  color?: string
  enabled: number
  remark?: string
  createdAt: string
  updatedAt: string
}

export interface PlatformCreateDTO {
  name: string
  color?: string
  enabled?: number
  remark?: string
}

export interface PlatformUpdateDTO {
  name: string
  color?: string
  enabled?: number
  remark?: string
}

// 分页查询平台列表
export function getPlatformList(params?: PageParams & { keyword?: string; enabled?: number }) {
  return request.get<any, ApiResponse<PageResult<Platform>>>('/platforms', { params })
}

// 获取平台详情
export function getPlatformDetail(id: number) {
  return request.get<any, ApiResponse<Platform>>(`/platforms/${id}`)
}

// 获取启用的平台列表（用于下拉选择）
export function getPlatformOptions() {
  return request.get<any, ApiResponse<Platform[]>>('/platforms/options')
}

// 创建平台
export function createPlatform(data: PlatformCreateDTO) {
  return request.post<any, ApiResponse<number>>('/platforms', data)
}

// 更新平台
export function updatePlatform(id: number, data: PlatformUpdateDTO) {
  return request.put<any, ApiResponse<void>>(`/platforms/${id}`, data)
}

// 删除平台
export function deletePlatform(id: number) {
  return request.delete<any, ApiResponse<void>>(`/platforms/${id}`)
}

File: wms-backend/src/main/java/com/yiweilai/wms/platform/entity/Platform.java
package com.yiweilai.wms.platform.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 平台实体
 */
@Data
public class Platform {

    /** 平台ID */
    private Long id;

    /** 平台名称 */
    private String name;

    /** 展示颜色（十六进制） */
    private String color;

    /** 是否启用：1-启用 0-禁用 */
    private Integer enabled;

    /** 备注 */
    private String remark;

    /** 逻辑删除：0-未删 1-已删 */
    private Integer deleted;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
